### PR TITLE
docker-ext: Add files required to build docker ext

### PR DESCRIPTION
**Pull request by yolossn**
To Do:

- [ ] Add Docker login creds to repo secrets.

### CODE REVIEW

File: docker-extension/README.md
Comment (illume) [inline docker-extension/README.md:2]: Is there a URL you could put here for related documentation?

**Review by illume — COMMENTED** [review]
👍

File: docker-extension/README.md
Comment (illume) [inline docker-extension/README.md:7]: After this, is there somewhere the user can go to run the docker desktop headlamp?

(Sorry if it's obvious... I haven't installed docker desktop yet... will do).

**Comment by illume** [comment]
Is this still a draft? If not, let me know and I'll review it :)

**Comment by yolossn** [comment]
> Is this still a draft? If not, let me know and I'll review it :)

Have to add make recipe for building the docker extension and pushing the extension to dockerhub.

File: Makefile
Comment (yolossn) [inline Makefile:94]: this will tag the docker image as "headlamp-dd-extension". Up for better naming suggestions.

File: docker-extension/docker-compose.yml
Comment (yolossn) [inline docker-extension/docker-compose.yml:5]: I am confused if the GitHub action to build and push the extension should get auto-triggered on every release or be manually triggered. Since the image version has to be updated every time before the release and there is a high chance that this gets missed out.

Comment (illume) [reply docker-extension/docker-compose.yml:5]: I think we're aiming to /automate all the things/ eventually, so probably good to set it up for on release.

**Comment by illume** [comment]
I found this page with a checklist of things to do before submitting the extension: https://www.docker.com/products/extensions/submissions/

In it was a section on running "validate":  https://docs.docker.com/desktop/extensions-sdk/build/build-install/#validate-your-extension

```bash
$ docker extension validate headlamp
label "com.docker.extension.additional-urls" is not specified. You can provide more detailed information to users about your extension.
could not parse image into name and tag
```

I'm wonder if there should be a make target for validate as well? Or maybe just in the README.md?

**Comment by illume** [comment]
After I installed it via the command line (gitbash) on windows it doesn't show up in docker desktop. hrmm.

image

**Comment by illume** [comment]
When I click to open the Headlamp extension it doesn't seem to do anything. Just a blank screen.

Then I clicked on the left hand icon again... and up pops Headlamp. But is stuck on "fetching clusters".
image

I'm not sure how to debug it or get logs for it... do you know?

**Comment by illume** [comment]
Part of the extention check list is following their design and UI guidelines. https://www.docker.com/products/extensions/submissions/

In there they talk about using mui (we are still on material ui 4) with their theme. So probably before we submit it we should do that. We have that mui v5 upgrade PR still open here: https://github.com/kinvolk/headlamp/pull/668 (it's still a bit buggy but 'works').

File: docker-extension/docker-compose.yml
Comment (yolossn) [reply docker-extension/docker-compose.yml:5]: In that case, we will have to make the updating the tag and create a commit also as a step in GitHub action.

Comment (yolossn) [reply docker-extension/docker-compose.yml:5]: let me check how feasible it is.

**Comment by yolossn** [comment]
> When I click to open the Headlamp extension it doesn't seem to do anything. Just a blank screen.
> 
> Then I clicked on the left hand icon again... and up pops Headlamp. But is stuck on "fetching clusters". 


Enable the kubernetes cluster from docker desktop settings before installing the headlamp addon, then it wont get stuck in "fetching clusters" stage. (Added this step to Readme)

image

**Comment by yolossn** [comment]
> Part of the extention check list is following their design and UI guidelines. https://www.docker.com/products/extensions/submissions/

I am not sure if this is compulsory for submitting the extension, I don't see other extensions using the docker desktop theme. So I think we can solve this once #668 is merged.

**Comment by illume** [comment]
> Enable the kubernetes cluster from docker desktop settings before installing the headlamp addon, then it won't get stuck in "fetching clusters" stage. (Added this step to Readme)

Cool, thanks. I tried turning k8s on and off again. Restarting docker desktop. Uninstalling headlamp extension... and now the extension won't install... the docker extension install command just hangs. I'm not having much luck here.

hrmm. Not sure what else to try.

> I am not sure if this is compulsory for submitting the extension, I don't see other extensions using the docker desktop theme. So I think we can solve this once https://github.com/kinvolk/headlamp/pull/668 is merged.

Yeah, that was my thinking too. Submit it and see what they say. It's already matching pretty close I guess using material ui and such.

File: Makefile
Comment (illume) [reply Makefile:94]: Seems fine to me. I guess headlamp-docker-desktop-extension was too long?

File: docker-extension/docker-compose.yml
Comment (illume) [reply docker-extension/docker-compose.yml:5]: This part could be done later? Joaquim has a better overview of the release process... and also where it is going.

**Review by illume — APPROVED** [review]
🎉

It's in good shape to merge already IMHO.

File: docker-extension/docker-compose.yml
Comment (yolossn) [reply docker-extension/docker-compose.yml:5]: Added steps to update the tag automatically and commit.
sharing a test run from my fork for reference:
https://github.com/yolossn/headlamp/runs/8224275472?check_suite_focus=true

Comment (yolossn) [reply docker-extension/docker-compose.yml:5]: > This part could be done later? Joaquim has a better overview of the release process... and also where it is going.

Have added the steps as discussed, I think we can wait till Joaquim is back to get feedback.

File: Makefile
Comment (yolossn) [reply Makefile]: >  I guess headlamp-docker-desktop-extension was too long

yup

File: docker-extension/Dockerfile
Comment (joaquimrocha) [inline docker-extension/Dockerfile:6]: Please remove the "brought to you by Kinvolk".

File: .github/workflows/docker-extension-release.yml
Comment (joaquimrocha) [inline .github/workflows/docker-extension-release.yml:48]: Should we push the changes to our repo in the `Commit changes...` step only after this step is successful?

**Review by joaquimrocha — CHANGES_REQUESTED** [review]
Left a couple of small comments.

**Pull request merged**
